includes/push-ui.php

```diff
@@ -88,7 +88,52 @@ function ajax_push() {
 		exit;
 	}
 
-	$connection_map = get_post_meta( intval( $_POST['postId'] ), 'dt_connection_map', true );
+	// Maybe we need $_POST stored for background task
+	$params = $_POST;
+
+	if ( ! wp_doing_cron() ) {
+		/**
+		 * Add possibility to send notification in background
+		 *
+		 * @param bool  false   Whether run 'push' action in background or not, default 'false'
+		 * @param array $params request data
+		 */
+		$push_in_background = apply_filters( 'dt_push_allow_in_background', false, $params );
+
+		if ( true === $push_in_background ) {
+			wp_send_json_success(
+				array(
+					'results' => 'Success!!',
+				)
+			);
+
+			exit;
+		}
+	}
+
+	$result = push( $params );
+
+	wp_send_json_success(
+		array(
```

Comment:
can we have `push` return the exact array we need and return it here directly?

Reply:
Changed accordingly.